safehttp/header_test.go

```diff
@@ -0,0 +1,200 @@
+		t.Errorf(`h.Values("Set-Cookie") mismatch (-want +got):\n%s`, diff)
+	}
+}
+
+func TestAddImmutable(t *testing.T) {
+	h := newHeader(http.Header{})
+	if err := h.Add("Foo-Key", "Bar-Value"); err != nil {
+		t.Fatalf(`h.Add("Foo-Key", "Bar-Value") got err: %v want: nil`, err)
+	}
+	h.MarkImmutable("Foo-Key")
+	err := h.Add("Foo-Key", "Bar-Value-2")
+	if got, want := err.Error(), "immutable header"; got != want {
+		t.Errorf(`h.Add("Set-Cookie", "Bar-Value") got: %v want: %v`, got, want)
+	}
+	if diff := cmp.Diff([]string{"Bar-Value"}, h.Values("Foo-Key")); diff != "" {
+		t.Errorf(`h.Values("Foo-Key") mismatch (-want +got):\n%s`, diff)
+	}
+}
+
+func TestDel(t *testing.T) {
+	h := newHeader(http.Header{})
+	if err := h.Set("Foo-Key", "Bar-Value"); err != nil {
+		t.Fatalf(`h.Set("Foo-Key", "Bar-Value") got err: %v want: nil`, err)
+	}
+	if err := h.Del("Foo-Key"); err != nil {
+		t.Fatalf(`h.Del("Foo-Key") got err: %v want: nil`, err)
+	}
+	if diff := cmp.Diff([]string{}, h.Values("Foo-Key")); diff != "" {
+		t.Errorf(`h.Values("Foo-Key") mismatch (-want +got):\n%s`, diff)
+	}
+}
+
+func TestDelCanonicalization(t *testing.T) {
+	h := newHeader(http.Header{})
+	if err := h.Set("fOo-KeY", "Bar-Value"); err != nil {
+		t.Fatalf(`h.Set("fOo-KeY", "Bar-Value") got err: %v want: nil`, err)
+	}
+	if err := h.Del("FoO-kEy"); err != nil {
+		t.Fatalf(`h.Del("FoO-kEy") got err: %v want: nil`, err)
+	}
+	if diff := cmp.Diff([]string{}, h.Values("FOO-kEY")); diff != "" {
+		t.Errorf(`h.Values("FOO-kEY") mismatch (-want +got):\n%s`, diff)
+	}
+}
+
+func TestDelSetCookie(t *testing.T) {
+	h := newHeader(http.Header{})
+	err := h.Del("Set-Cookie")
+	if got, want := err.Error(), "can't write to Set-Cookie header"; got != want {
+		t.Errorf(`h.Del("Set-Cookie") got: %v want: %v`, got, want)
+	}
+}
+
+func TestDelImmutable(t *testing.T) {
+	h := newHeader(http.Header{})
+	if err := h.Set("Foo-Key", "Bar-Value"); err != nil {
+		t.Fatalf(`h.Set("Foo-Key", "Bar-Value") got err: %v want: nil`, err)
+	}
+	h.MarkImmutable("Foo-Key")
+	err := h.Del("Foo-Key")
+	if got, want := err.Error(), "immutable header"; got != want {
+		t.Errorf(`h.Add("Set-Cookie", "Bar-Value") got: %v want: %v`, got, want)
+	}
+	if diff := cmp.Diff([]string{"Bar-Value"}, h.Values("Foo-Key")); diff != "" {
+		t.Errorf(`h.Values("Foo-Key") mismatch (-want +got):\n%s`, diff)
+	}
+}
+
+func TestSetCookie(t *testing.T) {
+	h := newHeader(http.Header{})
+	c := &http.Cookie{Name: "x", Value: "y"}
+	h.SetCookie(c)
+	if got, want := h.Get("Set-Cookie"), "x=y"; got != want {
+		t.Errorf(`h.Get("Set-Cookie") got: %q want: %q`, got, want)
+	}
+}
+
+func TestSetCookieInvalidName(t *testing.T) {
+	h := newHeader(http.Header{})
+	c := &http.Cookie{Name: "x=", Value: "y"}
+	h.SetCookie(c)
+	if got, want := h.Get("Set-Cookie"), ""; got != want {
+		t.Errorf(`h.Get("Set-Cookie") got: %q want: %q`, got, want)
+	}
+}
+
+func TestValuesModifyImmutable(t *testing.T) {
```

Comment:
Add a comment explaining that you are testing whether `Values()` returns a copy of a slice. Also, add a godoc comment mentioning this for `Values()`.